static/app/views/automations/components/automationBuilderContext.tsx

```diff
@@ -0,0 +1,257 @@
+          };
+        }
+
+        case 'ADD_IF_CONDITION': {
+          const {groupIndex, conditionType} = action;
+          return {
+            ...state,
+            actionFilters: state.actionFilters.map((group, i) => {
+              if (i !== groupIndex) {
+                return group;
+              }
+              return {
+                ...group,
+                conditions: [
+                  ...group.conditions,
+                  {
+                    comparison_type: conditionType as DataConditionType,
+                    comparison: {},
+                    condition_result: undefined,
+                  },
+                ],
+              };
+            }),
+          };
+        }
+
+        case 'REMOVE_IF_CONDITION': {
+          const {groupIndex, conditionIndex} = action;
+          if (formModel) {
+            for (const key of formModel.fields.keys()) {
+              if (
+                key.startsWith(
+                  `actionFilters.${groupIndex}.conditions.${conditionIndex}.`
+                )
+              ) {
+                formModel.removeField(key);
+              }
+            }
+          }
+          return {
+            ...state,
+            actionFilters: state.actionFilters.map((group, i) => {
+              if (i !== groupIndex) {
+                return group;
+              }
+              return {
+                ...group,
+                conditions: group.conditions.filter((_, j) => j !== conditionIndex),
+              };
+            }),
+          };
+        }
+
+        case 'UPDATE_IF_CONDITION': {
+          const {groupIndex, conditionIndex, comparison} = action;
+          return {
+            ...state,
+            actionFilters: state.actionFilters.map((group, i) => {
+              if (i !== groupIndex) {
+                return group;
+              }
+              return {
+                ...group,
+                conditions: group.conditions.map((c, j) =>
+                  j === conditionIndex
+                    ? {...c, comparison: {...c.comparison, ...comparison}}
+                    : c
+                ),
+              };
+            }),
+          };
+        }
+
+        case 'UPDATE_IF_LOGIC_TYPE': {
+          const {groupIndex, logicType} = action;
+          return {
+            ...state,
+            actionFilters: state.actionFilters.map((group, i) =>
+              i === groupIndex ? {...group, logicType} : group
+            ),
+          };
+        }
+
+        case 'UPDATE_WHEN_LOGIC_TYPE': {
+          const {logicType} = action;
+          return {
+            ...state,
+            triggers: {
+              ...state.triggers,
+              logicType,
+            },
+          };
+        }
+
+        default:
+          return state;
+      }
+    },
+    []
+  );
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  return {state, dispatch};
```

Comment:
As an alternative to exposing `dispatch` directly, it might be a nice DX to decompose these into individual action callbacks that can be picked from the context like `updateIfCondition({ groupIndex, conditionIndex, comparison })`.

Reply:
something like what i've done here? https://github.com/getsentry/sentry/pull/89233/commits/90a81f493480e9771d59074635dad5ed0021631e

Reply:
yep that's exactly what I was suggesting! nice one